src/components/features/PairwiseSearch/FeatureInfoPlot.vue

```diff
@@ -0,0 +1,220 @@
+            for(let result of this.results){
+              let descType = dataDesc2Type.get(result.dataDesc.id);
+              if(descType === undefined) descType = "unknown";
+
+              let geneRelationship = dataType2pairwiseRelationship.get(descType);
+              if(geneRelationship === undefined){
+                geneRelationship = {relationshipType: [], posNum: [], text: []};
+                dataType2pairwiseRelationship.set(descType, geneRelationship);
+              }
+
+                geneRelationship.relationshipType.push(result.dataDesc.id);
+                //TODO: case where no posNum
+                let posNum = result.posNum ? result.posNum: 0;
+                let negNum = result.negNum ? result.negNum: 0;
+                geneRelationship.posNum.push(posNum + negNum);
+                geneRelationship.text.push(this.generateText(result.posNum, result.dataDesc.id));
+            }
+          }
+
+          let plotData = [];
+          let relationshipTypes = dataType2pairwiseRelationship.keys();
+          for(let relationshipType of relationshipTypes) {
+            let pairwiseData = dataType2pairwiseRelationship.get(relationshipType);
+            let name = relationshipType;
+
+            let relationshipGeneData = {
+              name: name,
+              x: pairwiseData.relationshipType,
+              y: pairwiseData.posNum,
+              text: pairwiseData.text,
+              type: "scattergl",
+              mode: "markers",
+              marker: {
+                size: 5,
+              }
+            }
+            plotData.push(relationshipGeneData);
+          }
+          return plotData;
+        },
+  
+        generateText(numFeatures, relationship) {
+          if(relationship){
+          let relationships = relationship.split("|");
+          let relationshipType = relationships[2];
+          let dataResouce = relationships[0];
+          let bioSource = relationships[1];
+          let dataSource = relationships[3] ? relationships[3] : undefined;
+            let text = "<b>Relationship Type: " + relationshipType + "</b><br>" 
+            + "Data Resource: " + dataResouce + "<br>"
+            + "Biosource: " + bioSource;
+            if (dataSource) text += "<br>" + "Data Source: " + dataSource;
+            text = text + "<br>" + "Number of Features: " + numFeatures;
+            return text;
+          }
+        },
+  
+         getDataDescriptions() {
+            let data_descriptions_text = sessionStorage.getItem("dataDescriptions");
+            let tmpDataDesc= undefined;
+            if (data_descriptions_text) {
+                tmpDataDesc = JSON.parse(data_descriptions_text);
+            } else {
+                tmpDataDesc = "TODO"
+            }
+            let dataDescriptions = tmpDataDesc;
+
+            return dataDescriptions;
+        },
+
+        getDatadescriptionIds() {
+          this.dataDescriptions = this.getDataDescriptions();
```

Comment:
Would it be better to check if this.dataDescriptions is here already to avoid calling it again and again though it should be trivial?